pandas/core/arrays/datetimes.py

```diff
@@ -824,25 +825,40 @@ def _add_offset(self, offset: BaseOffset) -> Self:
                 "s",
             ]
             res_unit = self.unit
-            if hasattr(offset, "offset"):
-                offset_td = Timedelta(offset.offset)
-                offset_unit = offset_td.unit
-                if self.unit in units and offset_unit in units:
-                    idx_self = units.index(self.unit)
-                    idx_offset = units.index(offset_unit)
-                    res_unit = units[min(idx_self, idx_offset)]
+            if isinstance(offset, Tick) or (
```

Comment:
what you had before here seemed much more reasonable.  i dont see any reason for this to be specific to CustomBusinessDay